clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/CommitApplicationEvent.java

```diff
@@ -18,17 +18,36 @@
 
 import org.apache.kafka.clients.consumer.OffsetAndMetadata;
 import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.utils.Timer;
 
 import java.util.Collections;
 import java.util.Map;
+import java.util.Optional;
 
 public class CommitApplicationEvent extends CompletableApplicationEvent<Void> {
 
+    /**
+     * Offsets to commit per partition.
+     */
     private final Map<TopicPartition, OffsetAndMetadata> offsets;
 
-    public CommitApplicationEvent(final Map<TopicPartition, OffsetAndMetadata> offsets) {
+    /**
+     * Timer to wait for a response, retrying on retriable errors. If not present, the request is
+     * triggered without waiting for a response or being retried.
+     */
+    private final Optional<Timer> timer;
```

Comment:
Hmm, we are sending a mutable object between threads that will then be accessed by both threads. Even if that works out in the current code, I'd still prefer to just send the deadline and instantiate a new timer in the background thread so that we don't run into hairy concurrency bugs later on. `Timer` isn't thread safe

Reply:
Good catch! Changed it to use timeouts instead. Intentionally in the same way that the `TopicMetadataApplicationEvent` that has just been merged does it, trying to start getting a consistent implementation of timeouts across all appEvents and managers that need it.